### PR TITLE
CR #203

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

File: tests/http/server.cpp [hunk 1]
```diff
@@ -354,17 +368,17 @@ BOOST_AUTO_TEST_CASE(registration)
     BOOST_CHECK( server.remove( foo ));
     BOOST_CHECK( !server.remove( foo ));
 
-    BOOST_CHECK( server.handlePUT( "foo", zeroeq::PUTPayloadFunc
-                                  ([]( const std::string& ) { return true; })));
-    BOOST_CHECK( !server.handlePUT( "foo", zeroeq::PUTPayloadFunc
-                                  ([]( const std::string& ) { return true; })));
+    BOOST_CHECK( server.handlePUT( "foo",
+                                  {[]( const std::string& ) { return true; }}));
```
Comment (rdumusc) [inline tests/http/server.cpp]: yes, even the {} are not needed here (4 occurences)

File: zeroeq/http/request.h [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ *                     Stefan.Eilemann@epfl.ch
+ */
+
+#pragma once
+
+namespace zeroeq
+{
+namespace http
+{
+enum class Method
```
Comment (rdumusc) [inline zeroeq/http/request.h]: +1, Method is more correct than Verb. Shouldn't the header be named method.h?

File: zeroeq/http/server.cpp [hunk 3]
```diff
@@ -240,18 +248,17 @@ class Server::Impl : public detail::Sender
...
-
-        RESTFunc futureFunc = [ func ]( const std::string& )
+        const auto futureFunc = [ func ]( const std::string& )
         {
             return make_ready_future( Response{ Code::OK, func() } );
         };
-        _methods[int(Verb::GET)][ endpoint ] = futureFunc;
+
+	    if( !handle( Method::GET, endpoint, futureFunc ))
+                return false;
```
Comment (rdumusc) [inline zeroeq/http/server.cpp]: wrong indentation

File: zeroeq/http/server.cpp [hunk 4]
```diff
@@ -373,16 +359,16 @@ class Server::Impl : public detail::Sender
         return Response{ Code::OK, it->second };
     }
 
-    void _processRequest( HTTPRequest& request, Verb verb )
+    void _processRequest( Request& request, const Method verb )
```
Comment (rdumusc) [inline zeroeq/http/server.cpp]: rename verb -> method, and even remove the second param? I see that it is only called once as:
_processRequest( *request, request->method );

File: zeroeq/http/types.h [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+enum class Method;
+
+/** HTTP PUT callback w/o payload, return reply success. */
+using PUTFunc = std::function< bool() > ;
+
+/** HTTP PUT callback w/ JSON payload, return reply success. */
+using PUTPayloadFunc = std::function< bool( const std::string& ) >;
+
+/** HTTP GET callback to return JSON reply. */
+using GETFunc = std::function< std::string() >;
```
Comment (rdumusc) [inline zeroeq/http/types.h]: Actually those functions are used exclusively by the Server, move them in server.h? I don't think we need this header. In any case, we should group them with RESTFunc and RESTPathFunction for consistency.

File: tests/http/server.cpp (hunk 1 above)
Comment (eile) [inline tests/http/server.cpp]: Overlooked ... will fix.

File: zeroeq/http/request.h (hunk 2 above)
Comment (eile) [inline zeroeq/http/request.h]: Only if we clean up response.h in the same way.

File: zeroeq/http/server.cpp (hunk 4 above)
Comment (eile) [inline zeroeq/http/server.cpp]: rename done

File: zeroeq/http/server.cpp (hunk 3 above)
Comment (eile) [inline zeroeq/http/server.cpp]: (cough)clang-format(cough)

File: zeroeq/http/request.h (hunk 2 above)
Comment (rdumusc) [inline zeroeq/http/request.h]: OK, now I see what you mean, if we put a Request struct in here then it would be the same as response.h. Good then.

**Pull request merged**